tests/performance/ecommerce_hybrid_search/dataprep/generate_feed_query.py

```diff
@@ -0,0 +1,372 @@
+    /search/
+    {"yql": "select * from sources * where userQuery()", "ranking.profile": "bm25", "query": "blue nike shoes"}
+    """
+    endpoint = "/search/"
+    # Validate number of queries requested is less than the total number of samples
+    if num_queries > len(df):
+        raise ValueError(
+            "Number of queries requested is greater than the total number of samples available."
+        )
+    # Sample the dataframe set random state for reproducibility
+    df = df.sample(num_queries, random_state=42)
+    if query_type == QueryTypeEnum.WEAK_AND:
+        base_parameters = {
+            "yql": "select * from product where userQuery()",
+            "ranking.profile": "bm25",
+            "presentation.summary": "minimal",
+        }
+    elif query_type == QueryTypeEnum.SEMANTIC:
+        base_parameters = {
+            "yql": "select * from product where ({targetHits:100}nearestNeighbor(embedding,q_embedding))",
+            "ranking.profile": "closeness",
+            "presentation.summary": "minimal",
+        }
+    elif query_type == QueryTypeEnum.HYBRID:
+        base_parameters = {
+            "yql": "select * from product where ({targetHits:10}nearestNeighbor(embedding,q_embedding)) or userQuery()",
+            "ranking.profile": "hybrid",
+            "presentation.summary": "minimal",
+        }
+    else:
+        raise ValueError("Invalid query type")
+    # Generate list of queries
+    queries = df["title"].apply(title_to_query).tolist()
+    if query_type != QueryTypeEnum.WEAK_AND:
+        # Get embeddings
+        embeddings = df["embedding"].tolist()
+        full_queries = [
+            {
+                **base_parameters,
+                "query": remove_quote_chars(query),
+                "input.query(q_embedding)": embedding.tolist(),
+            }
+            for query, embedding in zip(queries, embeddings)
+        ]
+    else:
+        full_queries = [
+            {**base_parameters, "query": remove_quote_chars(query)} for query in queries
+        ]
+    # Write alternating lines to data
+    data = ""
+    for query in full_queries:
+        data += endpoint + "\n"
+        data += json.dumps(query, ensure_ascii=True) + "\n"
+
+    cctx = zstd.ZstdCompressor()
+    compressed_data = cctx.compress(data.encode("utf-8"))
+    with open(query_save_path, "wb") as f:
+        f.write(compressed_data)
+
+
+def save_es_query_files_from_df(
+    df: pd.DataFrame,
+    query_save_path: Path,
+    num_queries: int,
+    query_type: QueryTypeEnum = QueryTypeEnum.WEAK_AND,
+) -> None:
+    """
+    Format needed by fbench (POST):
+    Ref: https://docs.vespa.ai/en/performance/vespa-benchmarking.html#prepare-queries
+    Two lines per query, example:
+
+    /search/
+    {"query": {"match": {"title": "small money clip leather wallet"}}}
+    /search/
+    {"query": {"match": {"title": "blue nike shoes"}}}
+    """
+    endpoint = "/search/"

```

Comment:
I don't think this is the correct endpoint for ES? The same applies for lines 243 and 245.